src/Tools/NodeDocumentationMarkdownGenerator/AssemblyHandler.cs

```diff
@@ -0,0 +1,387 @@
+            }
+
+            return descriptors;
+        }
+
+        private static List<FunctionDescriptor> GetFunctionDescriptorsFromDs(PathManager pathManager, string dsFilePath)
+        {
+            var descriptors = new List<FunctionDescriptor>();
+
+            var t = new ImportModuleHandler(new ProtoCore.Core(new ProtoCore.Options()));
+            var dsCodeNode = t.Import(dsFilePath, "", "");
+            var classNode = dsCodeNode.CodeNode.Body.OfType<ClassDeclNode>().FirstOrDefault();
+            var associativeNodes = classNode.Procedures;
+            var customizationFile = LibraryCustomizationServices.GetForAssembly(dsFilePath, pathManager);
+            foreach (var node in associativeNodes)
+            {
+                if (TryGetFunctionDescriptor(node, null, dsFilePath, classNode.ClassName, customizationFile, out ReflectionFunctionDescriptor des))
+                {
+                    descriptors.Add(des);
+                }
+            }
+            return descriptors;
+        }
+
+        private static void AddZeroTouchNodesToSearch(IEnumerable<FunctionGroup> functionGroups, ref NodeSearchModel nodeSearchModel)
+        {
+            foreach (var funcGroup in functionGroups)
+            {
+                AddZeroTouchNodeToSearch(funcGroup, ref nodeSearchModel);
+            }
+        }
+
+        private static void AddZeroTouchNodeToSearch(FunctionGroup funcGroup, ref NodeSearchModel nodeSearchModel)
+        {
+            foreach (var functionDescriptor in funcGroup.Functions)
+            {
+                if (functionDescriptor.IsVisibleInLibrary)
+                {
+                    nodeSearchModel.Add(new ReflectionZeroTouchSearchElement(functionDescriptor as ReflectionFunctionDescriptor));
+                }
+            }
+        }
+
+        private static void FileInfosFromNodeModels(Assembly asm, Type nodeModelType, ref NodeSearchModel searchModel)
+        {
+
+            var fileInfos = new List<MdFileInfo>();
+            System.Type[] typesInAsm = null;
+            try
+            {
+                typesInAsm = asm.GetTypes();
+            }
+            catch (ReflectionTypeLoadException ex)
+            {
+                typesInAsm = ex.Types;
+            }
+
+            var nodeTypes = typesInAsm
+                .Where(x => NodeModelAssemblyLoader.IsNodeSubTypeReflectionLoaded(x, nodeModelType))
+                .Select(t => new TypeLoadData(t, t.GetAttributesFromReflectionContext()))
+                .Where(type => type != null && !type.IsDeprecated && !type.IsHidden)
+                .ToList();
+
+            foreach (var type in nodeTypes)
+            {
+                searchModel.Add(new NodeModelSearchElement(type, false));
+            }
+        }
+
+        private static List<string> GetDefaultPaths()
+        {
+            string[] runtimeAssemblies = Directory.GetFiles(RuntimeEnvironment.GetRuntimeDirectory(), "*.dll", SearchOption.AllDirectories);
```

Comment:
why do you need the .net assemblies?

Reply:
we essentially need them to create the MetadataLoadContext

https://docs.microsoft.com/en-us/dotnet/standard/assembly/inspect-contents-using-metadataloadcontext#create-metadataassemblyresolver-for-metadataloadcontext